### PR TITLE
uORB::Publication simplify and cleanup

### PR DESCRIPTION
**Pull request by dagar**
- drop linked list
 - base class is now template
 - virtualization no longer required
 - type safety!
 - ekf2 updated to only publish with uORB::Publication

### CODE REVIEW

**Comment by dannyfpv** [comment]
Tested on Pixhawk 4 V5 f-450
Modes Tested

Position Mode: no issue
Altitude Mode: no issue
Stabilized Mode: no issue
Mission Plan Mode: switch to hold mode
RTL:  switch to hold mode 

Notes:
no mission mode and no rtl mode upload to master firmware and no issues

pr tested log:
https://review.px4.io/plot_app?log=b93bbe2b-d7b4-409a-97c6-3e00998db9ca

**Comment by dagar** [comment]
Your home position wasn't being published. Can you please try again with the current state of the PR?
Thanks

**Comment by dannyfpv** [comment]
I'm wait 10 minutes for GPS Lock on the pr, change to master and have GPS Lock on 30 seconds, I'm flight again for confirming

**Comment by Junkim3DR** [comment]
### Tested on Pixhawk 4 mini v5

**Modes Tested**
- Position Mode: Good.
- Altitude Mode: Good.
- Stabilized Mode: Good.
- Mission Plan Mode (Automated): Good.
- RTL: Good.

**Procedure**
Arm and Take off in position mode, after flying for approximately one minute, switched to altitude then stabilized mode proceed to switch to mission plan mode then make sure that vehicle follows all waypoints as shown in QGC, once completed all waypoin activate RTL and see landing behaviour.

**Notes:**
No issues noted, good flight in general.

**Logs:**
_PR 12210_
- https://review.px4.io/plot_app?log=0e8905d1-97cb-4178-a039-cca3fc19dc50

_Master_
- https://review.px4.io/plot_app?log=b014c23c-c92a-45fd-8bc7-062bac5d909e

**Comment by dannyfpv** [comment]
Tested on Pixhawk 4 V5
Modes Tested

Position Mode: Good.
Altitude Mode: Good.
Stabilized Mode: Good.
Mission Plan Mode (Automated): Good.
RTL: Good.

- Procedure
Arm and Take off in stabilized mode, position mode, altitude, and mission mode

Notes:
No issues noted good flight in general.

pr Log:

https://review.px4.io/plot_app?log=2c809d10-c7f7-49bd-b278-c7efa225bf9c

master log:

https://review.px4.io/plot_app?log=71a5aa74-1e6e-4c9c-ae20-2baff55f3daf

File: src/modules/local_position_estimator/BlockLocalPositionEstimator.cpp [hunk 1]
```diff
@@ -47,14 +47,6 @@ BlockLocalPositionEstimator::BlockLocalPositionEstimator() :
 	_sub_sonar(nullptr),
 	_sub_landing_target_pose(ORB_ID(landing_target_pose), 1000 / 40, 0, &getSubscriptions()),
 	_sub_airdata(ORB_ID(vehicle_air_data), 0, 0, &getSubscriptions()),
-
-	// publications
-	_pub_lpos(ORB_ID(vehicle_local_position), -1, &getPublications()),
-	_pub_gpos(ORB_ID(vehicle_global_position), -1, &getPublications()),
-	_pub_odom(ORB_ID(vehicle_odometry), -1, &getPublications()),
-	_pub_est_status(ORB_ID(estimator_status), -1, &getPublications()),
-	_pub_innov(ORB_ID(ekf2_innovations), -1, &getPublications()),
```
Comment (julianoes) [inline src/modules/local_position_estimator/BlockLocalPositionEstimator.cpp]: Where did these go?

Comment (dagar) [reply src/modules/local_position_estimator/BlockLocalPositionEstimator.cpp]: Into BlockLocalPositionEstimator.hpp : https://github.com/PX4/Firmware/pull/12210/files#diff-a072a0a21e8690b891c1be60260d1bc7R269

**Comment by jorge789** [comment]
### Tested on PixRacer V4
**Modes Tested**

Position Mode: Good.
Stabilized Mode: Good.
Mission Plan Mode (Automated): Good.
RTL: Good.
**Procedure**
Arm and Take off in position mode, after flying for approximately one minute,then stabilized mode proceed to switch to mission plan mode then make sure that vehicle follows all waypoints as shown in QGC, once completed all waypoin activate RTL and see landing behaviour.

**Notes:**
No issues noted, good flight in general.

**Logs:**
https://review.px4.io/plot_app?log=e6d7dc6a-5843-461c-8931-630b18e1634b

MASTER COMPARISON
https://review.px4.io/plot_app?log=8705340c-1489-47db-adf4-81e32bf1fd55

File: src/modules/local_position_estimator/BlockLocalPositionEstimator.cpp (hunk 1 above)
Comment (julianoes) [reply src/modules/local_position_estimator/BlockLocalPositionEstimator.cpp]: Ah that's just the ctor!

**Pull request merged**